src/arm/soc.c

```diff
@@ -14,6 +14,23 @@
   #include "../common/sysctl.h"
 #endif
 
+#if defined(_WIN32)
+  #define WIN32_LEAN_AND_MEAN
+  #define NOMINMAX
+  #include <windows.h>
+
+bool read_registry_hklm_sz(char* path, char* value, char** string, LPDWORD length) {
```

Comment:
#Could you add a comment to elaborate what this function does? Seems similar to `read_registry_hklm_int` but with strings, a comment explaining this would be useful. I'm also confused, why do you call `RegGetValueA` twice? You get the string length first, then you malloc, then you get the actual string? Explaining this in the comment as well would be beneficial.